networkx/algorithms/operators/product.py

```diff
@@ -432,3 +434,41 @@ def power(G, k):
             level += 1
         H.add_edges_from((n, nbr) for nbr in seen)
     return H
+
+def rooted_product(G, H, root):
+    """ Return the rooted product of graphs G and H rooted at root in H.
+
+    Parameters
+    ----------
+    G,H : graph
+       A NetworkX graph
+    root : node
+       A node in H    
+
+    Returns
+    -------
+    R : The rooted product of G and H with a specified root in H
+
+    Notes
+    -----
+    A new graph is constructed representing the rooted product of
+    the inputted graphs, G and H, with a root in H.
+
+    The nodes of R are the Cartesian Product of the nodes of G and H. The
+    nodes of G and H are not relabeled.
+
+    """
+
+    if G.is_multigraph() or H.is_multigraph():
+        raise nx.NetworkXError('G and H must not be multigraphs')
+
+    if root not in H:
+        raise nx.NetworkXError('root must be a vertex in H')
+
+    R = nx.Graph()
+    R.add_nodes_from(product(G,H))
+    
+    R.add_edges_from([((e[0],root),(e[1],root)) for e in G.edges()] 
```

Comment:
This is probably clearer as the following two lines:

    R.add_edges_from(((u, root), (v, root)) for u, v in G.edges())
    R.add_edges_from(((u, w), (v, w)) for w in G for u, v in H.edges())

Note the lack of square brackets: this is a generator expression, so we don't need to create the full list in memory, which may be very expensive in terms of space for large graphs.